python/fnllm/fnllm/events/composite.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2024 Microsoft Corporation.
+
+"""Class for LLM composite event handling."""
+
+import asyncio
+from collections.abc import Sequence
+from typing import Any
+
+from fnllm.events.base import LLMEvents
+from fnllm.limiting.base import Manifest
+from fnllm.types.metrics import LLMMetrics, LLMUsageMetrics
+
+
+class LLMCompositeEvents(LLMEvents):
```

Comment:
We could support a protocol, but the base class provides some advantages: you have good feedback if your methods drift from the base signature, and you don't have to implement all the event methods.

Reply:
Yup, I was thinking we could use protocols so that we don't have to implement all the methods defined in the base protocol if we don't need them and still be compliant with it.